### DISCUSSION TITLE
Drag&Drop and Popup

### DISCUSSION THREAD

**Discussion by NOoBODDY** [AvaloniaUI/Avalonia#14680] [Q&A] [answered]
I would like to show MenuFlyot in drop position/ But when i show it at drop event, it shows in start drag position.
As i think, this problem is related to the lack of pointer events during drag and drop.  Because PopupHost uses topLevel.LastPointerPosition for its position.

I create test rep for reproduce problem: https://github.com/NOoBODDY/DragNDropTest

**Comment by timunie** [comment, marked as answer]
```cs
Dispatcher.UIThread.Post(() => { menu.ShowAt(canvas, true); }, DispatcherPriority.Background); 
```

this solves the issue for me. The thing is that your behavior captures the pointer and we need to wait until it is returned before showing the MenuFlyout.
